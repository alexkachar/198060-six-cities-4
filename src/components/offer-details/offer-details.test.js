import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import OfferDetails from './offer-details.jsx';
import {BrowserRouter} from 'react-router-dom';

import mockStore from '../../mocks/mock-store.js';
import mockOffers from '../../mocks/offers.js';

it(`OfferDetails renders correctly`, () => {
  const tree = renderer
    .create(<Provider store={mockStore}>
      <BrowserRouter>
        <OfferDetails offer={mockOffers[0]} />
      </BrowserRouter>
    </Provider>, {
      createNodeMock: () => document.createElement(`div`)
    })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
