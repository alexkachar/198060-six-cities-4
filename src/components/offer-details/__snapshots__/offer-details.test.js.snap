// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferDetails renders correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height={41}
              src="/img/logo.svg"
              width={81}
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    src="https://4.react.pages.academy/six-cities/static/avatar/8.jpg"
                    style={
                      Object {
                        "borderRadius": "50%",
                      }
                    }
                  />
                </div>
                <span
                  className="header__user-name user__name"
                >
                  mail@email.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="hotel photo"
              className="property__image"
              src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="hotel photo"
              className="property__image"
              src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg,"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Penthouse, 4-5 rooms + 5 balconies
            </h1>
            <button
              className="button property__bookmark-button"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height={33}
                style={
                  Object {
                    "fill": "none",
                    "stroke": "#979797",
                  }
                }
                width={31}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.4
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              hotel
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              8
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              248
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height={74}
                  src="/img/avatar-angelina.jpg"
                  width={74}
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper null"
                  >
                    <img
                      alt="Max avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="img/avatar-max.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Max
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        4
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-04-24"
                  >
                    April 24, 2019
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper null"
                  >
                    <img
                      alt="Angelina avatar"
                      className="reviews__avatar user__avatar"
                      height={54}
                      src="img/avatar-angelina.jpg"
                      width={54}
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Angelina
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        5
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2019-05-14"
                  >
                    May 14, 2019
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={5}
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={4}
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={3}
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not good"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={2}
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  defaultValue={1}
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height={33}
                    width={37}
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                defaultValue=""
                id="review"
                maxLength={300}
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  . Maximum review length is 
                  <b
                    className="reviews__text-amount"
                  >
                    300 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
    </section>
    <section
      className="map property__map"
      id="map"
    />
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="place-card near-places__card"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="place-card__image-wrapper near-places__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €248
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating 
                    4.4
                  </span>
                </div>
              </div>
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                <h2
                  className="place-card__name"
                >
                  Penthouse, 4-5 rooms + 5 balconies
                </h2>
              </a>
              <p
                className="place-card__type"
              >
                hotel
              </p>
            </div>
          </article>
          <article
            className="place-card near-places__card"
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="place-card__image-wrapper near-places__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height={200}
                  src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
                  width={260}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €248
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating 
                    4.4
                  </span>
                </div>
              </div>
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                <h2
                  className="place-card__name"
                >
                  Penthouse, 4-5 rooms + 5 balconies
                </h2>
              </a>
              <p
                className="place-card__type"
              >
                hotel
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
